internal/apigw/templates.go

```diff
@@ -60,7 +60,8 @@ type varNamer struct {
 }
 
 func (fn *varNamer) Next() *varNamer {
-	return &varNamer{offset: fn.offset + 1, prefix: fn.prefix}
+	fn.offset += 1
```

Comment:
i'm not sure this is safe?  this doesnt' return a new varnamer?

Reply:
I actually think the copy-based namer has unexpected behavior:

```
package main

import (
	"fmt"
	"strconv"
)

type updateNextVN struct {
	val int
	prefix string
}

type copyNextVN struct {
	val int
	prefix string
}

type VarNamer interface {
	Next() VarNamer
	String() string
}

func (u *updateNextVN) Next() VarNamer {
	u.val++
	return u
}

func (u *updateNextVN) String() string {
	return u.prefix + strconv.Itoa(u.val)
}

func (c *copyNextVN) Next() VarNamer {
	return &copyNextVN{
		val: c.val + 1,
		prefix: c.prefix,
	}
}

func (c *copyNextVN) String() string {
	return c.prefix + strconv.Itoa(c.val)
}

func main() {
	copyNamer := &copyNextVN{val: 0, prefix: "copy"}
	updateNamer := &updateNextVN{val: 0, prefix: "update"}

	funcA(copyNamer)
	funcA(updateNamer)
}


func funcA(vn VarNamer) {
	fmt.Println(vn.String())
	vn = vn.Next()
	fmt.Println(vn.String())

	funcB(vn)

	fmt.Println(vn.String())
	vn = vn.Next()
	fmt.Println(vn.String())
}

func funcB(vn VarNamer) {
	fmt.Println(vn.String())
	vn = vn.Next()
	fmt.Println(vn.String())
}
```

```
copy0
copy1
copy1
copy2
copy1
copy2
update0
update1
update1
update2
update2
update3
```